metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -444,6 +450,34 @@ class SnowflakePrivilege:
 
         return _report
 
+    def _is_temp_table(self, name: str) -> bool:
+        if any(
+            re.match(pattern, name, flags=re.IGNORECASE)
+            for pattern in self.config.temporary_tables_pattern
+        ):
+            return True
+
+        # This is also a temp table if
```

Comment:
maybe note that this logic mirrors the logic from snowflake_queries.py?

not sure if there's a way to remove the code duplication, but that'd be even better

Reply:
This PR extends the existing note here
https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py#L545-L547
which is what causes the duplication

Reply:
The only thing that comes to my mind would be to give the responsibility of instantiating `SqlParsingAggregator` to the `SnowflakeV2Source` itself, so `SnowflakeQueriesSource` gets it as constructor parameter.
This way we still instantiate `SqlParsingAggregator` twice but `is_temp_table` and `is_allowed_table` would be defined once in the `SnowflaV2Source`
WDYT?

Reply:
I spent some time trying this, and it's getting complicated, so I'm leaving it as is.